crawl-ref/source/mutation.cc

```diff
@@ -935,53 +927,37 @@ static int _handle_conflicting_mutations(mutation_type mutation,
                                          const string &reason,
                                          bool temp = false)
 {
-    // If we have one of the pair, delete all levels of the other,
-    // and continue processing.
-    for (const int (&confl)[3] : conflict)
+    for (const mut_conflict conflict : mut_conflicts)
     {
-        for (int j = 0; j < 2; ++j)
+        if (conflict.muts[0] != mutation && conflict.muts[1] != mutation)
+            continue;
+
+        const mutation_type conf_mut = conflict.muts[0] == mutation ? conflict.muts[1]
+                                                                    : conflict.muts[0];
+
+        if (you.mutation[conf_mut] > 0)
         {
-            const mutation_type a = (mutation_type)confl[j];
-            const mutation_type b = (mutation_type)confl[1-j];
+            if (you.innate_mutation[conf_mut] >= you.mutation[conf_mut])
+                return -1;
 
-            if (mutation == a && you.mutation[b] > 0)
+            switch (override ? conflict.forced
+                    : temp   ? (conflict.unforced == MUTR_IGNORE ? MUTR_IGNORE : MUTR_COEXIST)
+                    :          conflict.unforced)
```

Comment:
For future, this sort of code should be broken out, probably into its own function. Something like:

`static mutation_resolution _get_mutation_response(const mut_conflict &conflict,
                                                  bool force, bool temp)
{
    if (force)
        return conflict.forced;
    if (!temp)
        return conflict.unforced;
    if (conflict.unforced == MUTR_IGNORE)
        return MUTR_IGNORE;
    return MUTR_COEXIST;
}`